### PR TITLE
Replace highest with lowest for signed epoch DB methods

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**

Studying the EIP again, it actually says `Refuse to sign any attestation with source epoch less than the minimum source epoch present in that signer’s attestations`. Which is different than what i had thought, we need to track the lowest signed epoch, not the highest signed epoch.

https://eips.ethereum.org/EIPS/eip-3076#conditions

**Which issues(s) does this PR fix?**

Fixes #7964 
**Other notes for review**

### CODE REVIEW

File: validator/db/kv/attestation_history_v2.go [hunk 1]
```diff
@@ -344,63 +344,63 @@ func (store *Store) shouldMigrateAttestations() (bool, error) {
...
 	return store.update(func(tx *bolt.Tx) error {
-		bucket := tx.Bucket(highestSignedSourceBucket)
+		bucket := tx.Bucket(lowestSignedSourceBucket)
 
-		// If the incoming epoch is higher than the highest signed epoch, override.
-		highestSignedSourceBytes := bucket.Get(publicKey[:])
-		var highestSignedSourceEpoch uint64
-		if len(highestSignedSourceBytes) >= 8 {
-			highestSignedSourceEpoch = bytesutil.BytesToUint64BigEndian(highestSignedSourceBytes)
+		// If the incoming epoch is higher than the lowest signed epoch, override.
```
Comment (rauljordan) [inline validator/db/kv/attestation_history_v2.go:397]:
```suggestion
		// If the incoming epoch is lower than the lowest signed epoch, override.
```

File: validator/db/kv/attestation_history_v2.go [hunk 2]
```diff
@@ -409,21 +409,21 @@ func (store *Store) SaveHighestSignedSourceEpoch(ctx context.Context, publicKey
...
 	return store.update(func(tx *bolt.Tx) error {
-		bucket := tx.Bucket(highestSignedTargetBucket)
+		bucket := tx.Bucket(lowestSignedTargetBucket)
 
-		// If the incoming epoch is higher than the highest signed epoch, override.
-		highestSignedTargetBytes := bucket.Get(publicKey[:])
-		var highestSignedTargetEpoch uint64
-		if len(highestSignedTargetBytes) >= 8 {
-			highestSignedTargetEpoch = bytesutil.BytesToUint64BigEndian(highestSignedTargetBytes)
+		// If the incoming epoch is higher than the lowest signed epoch, override.
```
Comment (rauljordan) [inline validator/db/kv/attestation_history_v2.go:420]:
```suggestion
		// If the incoming epoch is lower than the lowest signed epoch, override.
```

File: validator/slashing-protection/local/standard-protection-format/import.go [hunk 3]
```diff
@@ -259,34 +259,34 @@ func saveHighestSourceTargetToDB(ctx context.Context, validatorDB db.Database, s
 			if err != nil {
 				return fmt.Errorf("%d is not a valid target: %v", target, err)
 			}
-			se, ok := validatorHighestSourceEpoch[pubKey]
+			se, ok := validatorLowestSourceEpoch[pubKey]
 			if !ok {
-				validatorHighestSourceEpoch[pubKey] = source
+				validatorLowestSourceEpoch[pubKey] = source
 			} else if source > se {
```
Comment (rauljordan) [inline validator/slashing-protection/local/standard-protection-format/import.go:265]: This also needs to change

File: validator/db/iface/interface.go [hunk 4]
```diff
@@ -35,10 +35,10 @@ type ValidatorDB interface {
 	// Attester protection related methods.
 	AttestationHistoryForPubKeys(ctx context.Context, publicKeys [][48]byte) (map[[48]byte]*slashpb.AttestationHistory, error)
 	SaveAttestationHistoryForPubKeys(ctx context.Context, historyByPubKey map[[48]byte]*slashpb.AttestationHistory) error
-	HighestSignedTargetEpoch(ctx context.Context, publicKey [48]byte) (uint64, error)
```
Comment (nisdas) [inline validator/db/iface/interface.go:38]: Just to confirm , this method has been there this whole time ? `Highest` instead of `Lowest`

Comment (nisdas) [reply validator/db/iface/interface.go:38]: Ignore the comment

**Review by nisdas — DISMISSED** [review]

File: validator/slashing-protection/local/standard-protection-format/import.go [hunk 5]
```diff
@@ -267,34 +267,34 @@ func saveHighestSourceTargetToDB(ctx context.Context, validatorDB db.Database, s
 			if err != nil {
 				return fmt.Errorf("%d is not a valid target: %v", target, err)
 			}
-			se, ok := validatorHighestSourceEpoch[pubKey]
+			se, ok := validatorLowestSourceEpoch[pubKey]
```
Comment (rauljordan) [inline validator/slashing-protection/local/standard-protection-format/import.go:270]: I would rename these for clarity to better understand them in a separate PR. something like incomingAttSource and existingAttSource or something like that. no need to do in this one

**Review by rauljordan — APPROVED** [review]

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**